File: CodeEval/easy/004sumofprimes/solution.rb
#!/usr/bin/env ruby

def isprime(n)
  return false if n&1 == 0
  (3..Math.sqrt(n)).step(2) do |i|
    return false if n % i == 0
  end
  true
end

def primes(n)
  i = 1
  sum = 2
  x = 3
  until i >= n
    sum += x
    x += 2
    x += 2 until isprime(x)
    i += 1
  end
  sum
end

puts primes(1000)

File: codeeval/easy/004sumofprimes/solution.r

isprime <- function(n) n <= 3L || all(n %% 2L:floor(sqrt(n)) != 0)

primes <- function(n){
    sum <- 1
    x <- -1
    for (i in 1:n) {
        repeat {
            x <- x + 2
            if (isprime(x)) {
                break
            }
        }
        sum <- sum + x
    }
    sum
}

cat(primes(1000))
# cat(primes(5))
